pkg/judge/judge.go

```diff
@@ -0,0 +1,46 @@
+package judge
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/jasondellaluce/synchro/pkg/utils"
+)
+
+func Judge(ctx context.Context, git utils.GitHelper, commit string) error {
+	out, err := git.DoOutput("show", "--raw", commit)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(out, "\n")
+
+	modified := 0
+	added := 0
+	deleted := 0
+	renamed := 0
+
+	for _, line := range lines {
+		//only check file metadata lines
+		if len(line) > 0 && line[0] == ':' {
+			//file mode is the 32nd character of the line
```

Comment:
// file mode is the 32nd character of the line

Is it guaranteed? Can we check that 1) the line has at least 32 chars, and that the character is one of `M`, `A`, `D`, `R`, just to avoid painful debugging sessions in case something is wrong?

Reply:
I don't like this way of getting commit information (parsing/scraping `git show` output), but I think it's the only way.
So I agree! We should definitely make sure the output is in the expected form.